app/queries/keyset_pagination.rb

```diff
@@ -0,0 +1,130 @@
+module Queries
+  class KeysetPagination
+    attr_reader :client, :order, :key, :count, :previous
+
+    # Initialises the keyset pagination class.
+    # Params:
+    # +client+:: the pagination client to query over
+    # +key+:: +Hash+ a hash containing the pagination key, mapped from presented name to internal name, i.e. { id: "editions.id" }
+    # +order+:: +Symbol+ either :asc for ascending and :desc for descending
+    # +count+:: +Fixnum+ the number of records to return in each page
+    # +after+:: +Array+ the current page to paginate after, an array containing a value for each field in the key
```

Comment:
also might be good to rename key to pagination_key it gets a bit confusing seeing `key.keys`